back/lib/tasks/single_use/20250515_republish_traffic_sources_widgets.rake

```diff
@@ -0,0 +1,77 @@
+# Usage:
+#
+# Dry run: rake single_use:republish_traffic_sources_widgets
+# Execute: rake single_use:republish_traffic_sources_widgets[execute]
+namespace :single_use do
+  desc 'Republish traffic sources widgets'
+  task :republish_traffic_sources_widgets, %i[execute] => [:environment] do |_t, args|
+    execute = args[:execute] == 'execute'
+
+    def resolved_name(node)
+      type = node['type']
+      type['resolvedName'] if type.is_a?(Hash)
+    end
+
+    def republish_report(report, execute)
+      user = report.owner || User.super_admins.first || User.admins.first
+
+      if execute
+        ReportBuilder::ReportPublisher.new(report, user).publish
+      end
```

Comment:
Would some output in the dry-run case be useful to check things are happening as expected? e.g. in an `else` statement.